File: 0162-find-peak-element/0162-find-peak-element.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        n=len(nums)
        s=0
        e=n-1
        ans=0
        while(s<e):
            mid=s+(e-s)//2
            if(nums[mid]<nums[mid+1]):
                s=mid+1
            else:
    
                e=mid
        return s

        

File: 0162-find-peak-element/0162-find-peak-element.java
class Solution {
    public int findPeakElement(int[] nums) {
        int n=nums.length;
        int s=0;
        int e=n-1;
        int ans=0;
        while(s<e){
            int mid=(s+e)/2;
            if(nums[mid]<nums[mid+1]){
                s=mid+1;
                
            }else{
               
                e=mid;
            }
        }return s;
    }
}